Shared/Samples/Set up location-driven geotriggers/SetUpLocationDrivenGeotriggersView.swift

```diff
@@ -0,0 +1,332 @@
+            Task.detached { [unowned self] in
+                do {
+                    try await geotriggerMonitor.start()
+                    geotriggerMonitors.append(geotriggerMonitor)
+                } catch {
+                    self.error = error
+                }
+            }
+            
+            // Handle posted geotrigger notifications.
+            notificationTaskHandle = Task {
+                for await notification in geotriggerMonitor.notifications {
+                    handleGeotriggerNotification(notification as! FenceGeotriggerNotificationInfo)
+                }
+            }
+        }
+        
+        /// Handle a notification posted by a geotrigger monitor when a fence geotrigger
+        /// condition has been met.
+        /// - Parameter fenceNotificationInfo: The `FenceGeotriggerNotificationInfo`
+        /// which provides information about the geotrigger monitor and the fence
+        /// geotrigger that was triggered.
+        private func handleGeotriggerNotification(_ fenceNotificationInfo: FenceGeotriggerNotificationInfo) {
+            // The feature name from the Arcade expression.
+            let featureName = fenceNotificationInfo.message
+            let fenceFeature = fenceNotificationInfo.fenceGeoElement as! ArcGISFeature
+            let geotriggerName = fenceNotificationInfo.geotriggerMonitor.geotrigger.name
+            
+            // Handle notification types.
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+                switch fenceNotificationInfo.fenceNotificationType {
+                case .entered:
+                    // The user entered a geofence: add the feature for future querying.
+                    self.featureNamesInFenceGeotrigger[geotriggerName, default: []].append(featureName)
+                    self.nearbyFeatures[featureName] = fenceFeature
+                case .exited:
+                    // The user left the geofence: remove the feature from the dicts.
+                    if let poppedFeatureName = self.featureNamesInFenceGeotrigger[geotriggerName]?.popLast() {
+                        self.nearbyFeatures.removeValue(forKey: poppedFeatureName)
+                    }
+                @unknown default:
+                    fatalError("Unexpected fence notification type.")
+                }
+                
+                // Update the status texts.
+                self.updateStatusText(featureName: featureName, notificationType: fenceNotificationInfo.fenceNotificationType)
+            }
+        }
+        
+        /// Update the status texts with the feature name and notification type.
+        /// - Parameters:
+        ///   - featureName: The name of the feature.
+        ///   - notificationType: The fence `NotificationType`.
+        private func updateStatusText(featureName: String, notificationType: FenceGeotriggerNotificationInfo.NotificationType) {
```

Comment:
Put this method on the main actor. It would trigger a series of async context errors (so-called "async/await hell" in JavaScript). Try to restructure the logic. Leave it to me if you have troubles with it.
```suggestion
        @MainActor
        private func updateStatusText(featureName: String, notificationType: FenceGeotriggerNotificationInfo.NotificationType) {
```